coalib/results/Result.py

```diff
@@ -63,6 +65,13 @@ def __init__(self,
         if severity not in RESULT_SEVERITY.reverse:
             raise ValueError("severity is not a valid RESULT_SEVERITY")
 
+        self.diffs = {}
+        if(diffs and type(diffs) != dict):
+            for diff in diffs:
+                self.diffs[diff.filename] = diff
+        else:
+            self.diffs = diffs
```

Comment:
actually I think we should set this to an empty dict instead of None if None is given since using Nones often involves special handling whereas the empty dict really is sane.